File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/offroute/OffRouteDetector.java
package com.mapbox.services.android.navigation.v5.offroute;

import android.location.Location;

import com.mapbox.navigator.NavigationStatus;
import com.mapbox.navigator.RouteState;
import com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationOptions;
import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;

public class OffRouteDetector extends OffRoute {

  @Override
  public boolean isUserOffRoute(Location location, RouteProgress routeProgress, MapboxNavigationOptions options) {
    // No impl
    return false;
  }

  public boolean isUserOffRouteWith(NavigationStatus status) {
    return status.getRouteState() == RouteState.OFFROUTE;
  }
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/offroute/OffRouteDetector.kt
package com.mapbox.services.android.navigation.v5.offroute

import android.location.Location
import com.mapbox.navigator.NavigationStatus
import com.mapbox.navigator.RouteState
import com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationOptions
import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress

class OffRouteDetector : OffRoute() {

    override fun isUserOffRoute(
        location: Location,
        routeProgress: RouteProgress,
        options: MapboxNavigationOptions
    ): Boolean {
        // No impl
        return false
    }

    fun isUserOffRouteWith(status: NavigationStatus): Boolean {
        return status.routeState == RouteState.OFF_ROUTE
    }
}
